src/parser/core/modules/Medicated.ts

```diff
@@ -38,27 +38,36 @@ export class Medicated extends Analyser {
 		this.pot = {start: event.timestamp}
 	}
 
+	private stopAndSave(endTime: number) {
+		if (this.pot != null) {
+			this.pots.push({...this.pot, end: endTime})
+		}
+		this.pot = undefined
+	}
+
 	private offMedication(event: Events['statusRemove']) {
 		if (this.pot == null) {
 			throw new Error('potion instance not found')
 		}
-
-		this.pots.push({end: event.timestamp, ...this.pot})
-
-		this.pot = undefined
+		this.stopAndSave(event.timestamp)
 	}
 
 	private onComplete() {
-		const status = this.data.statuses.MEDICATED
+
```

Comment:
Can prob lose the whitespace here.

Reply:
Reverted. I've also added a newline between each block & statement.